File: output/lec-python-redo/course/lesson10/in_class/practice05.py
"""
[难度: ★★★][所属知识点: 结算与小票打印][预计完成时间: 15 分钟]

题目:实现 checkout 函数。
遍历购物车,打印每项小计,对齐输出,
最后打印结算总价,然后清空购物车。
"""

if __name__ == "__main__":
    print("参考答案:")
    print('def checkout(cart):')
    print('    if not cart:')
    print('        print("购物车为空")')
    print('        return')
    print('    print("=== 购物小票 ===")')
    print('    total = 0')
    print('    for item in cart:')
    print('        subtotal = item["price"] * item["qty"]')
    print('        total += subtotal')
    print('        print(f"  {item[\'name\']:8s} x{item[\'qty\']}  ¥{subtotal:.2f}")')
    print('    print(f"  总计: ¥{total:.2f}")')
    print('    cart.clear()')
    print('    print("结算成功!")')

File: output/iter-R9-js-build-pedagogy/course/lesson03/in_class/practice05.js
/*
[难度: ★★★]
[所属知识点: 综合场景选择]
[预计完成时间: 15 分钟]

题目:给定以下代码,把 var 改成 let/const,并说明每个改动的理由。

    function calculateTotal(items) {
        var total = 0;
        for (var i = 0; i < items.length; i++) {
            var price = items[i].price;
            total = total + price;
        }
        var TAX_RATE = 0.1;
        total = total * (1 + TAX_RATE);
        return total;
    }
*/

if (require.main === module) {
  console.log("修正后:");
  console.log("  function calculateTotal(items) {");
  console.log("      let total = 0;                    // 重新赋值");
  console.log("      for (let i = 0; i < items.length; i++) {  // 块级,不泄漏");
  console.log("          let price = items[i].price;     // 块级,每次循环新变量");
  console.log("          total = total + price;");
  console.log("      }");
  console.log("      const TAX_RATE = 0.1;             // 永远不变");
  console.log("      total = total * (1 + TAX_RATE);");
  console.log("      return total;");
  console.log("  }");
}
